src/internal/model/MetricsEventData.ts

```diff
@@ -0,0 +1,118 @@
+  SizeMetrics: 'type.googleapis.com/bucketeer.event.client.SizeMetricsEvent',
+  BadRequestError: 'type.googleapis.com/bucketeer.event.client.BadRequestErrorMetricsEvent',
+  UnauthorizedError: 'type.googleapis.com/bucketeer.event.client.UnauthorizedErrorMetricsEvent',
+  ForbiddenError: 'type.googleapis.com/bucketeer.event.client.ForbiddenErrorMetricsEvent',
+  NotFoundError: 'type.googleapis.com/bucketeer.event.client.NotFoundErrorMetricsEvent',
+  ClientClosedRequestError: 'type.googleapis.com/bucketeer.event.client.ClientClosedRequestErrorMetricsEvent',
+  InternalServerError: 'type.googleapis.com/bucketeer.event.client.InternalServerErrorMetricsEvent',
+  ServiceUnavailableError: 'type.googleapis.com/bucketeer.event.client.ServiceUnavailableErrorMetricsEvent',
+  InternalSdkError: 'type.googleapis.com/bucketeer.event.client.InternalSdkErrorMetricsEvent',
+  TimeoutError: 'type.googleapis.com/bucketeer.event.client.TimeoutErrorMetricsEvent',
+  NetworkError: 'type.googleapis.com/bucketeer.event.client.NetworkErrorMetricsEvent',
+  UnknownError: 'type.googleapis.com/bucketeer.event.client.UnknownErrorMetricsEvent',
+} as const
+
+export interface LatencyMetricsEvent {
+  apiId: ApiId
+  labels: Record<string, string>
+  duration: number
+  '@type': typeof MetricsEventType['LatencyMetrics']
+}
+
+export interface SizeMetricsEvent {
+  apiId: ApiId
+  labels: Record<string, string>
+  size_byte: number
+  '@type': typeof MetricsEventType['SizeMetrics']
+}
+
+// 400: Bad Request
+export interface BadRequestErrorMetricsEvent {
+  apiId: ApiId
+  labels: Record<string, string>
+  '@type': typeof MetricsEventType['BadRequestError']
+}
+
+// 401: Unauthorized
+export interface UnauthorizedErrorMetricsEvent {
+  apiId: ApiId
+  labels: Record<string, string>
+  '@type': typeof MetricsEventType['UnauthorizedError']
+}
+
+// 403: Forbidden
+export interface ForbiddenErrorMetricsEvent {
+  apiId: ApiId
+  labels: Record<string, string>
+  '@type': typeof MetricsEventType['ForbiddenError']
+}
+
+// 404: NotFound
+export interface NotFoundErrorMetricsEvent {
+  apiId: ApiId
+  labels: Record<string, string>
+  '@type': typeof MetricsEventType['NotFoundError']
+}
+
+// 499: Client Closed Request
+export interface ClientClosedRequestMetricsEvent {
+  apiId: ApiId
+  labels: Record<string, string>
+  '@type': typeof MetricsEventType['ClientClosedRequestError']
+}
+
+// 500: Internal Server Error
+export interface InternalServerErrorMetricsEvent {
+  apiId: ApiId
+  labels: Record<string, string>
+  '@type': typeof MetricsEventType['InternalServerError']
+}
+
+// 503: Service Unavailable
+export interface ServiceUnavailableErrorMetricsEvent {
+  apiId: ApiId
+  labels: Record<string, string>
+  '@type': typeof MetricsEventType['ServiceUnavailableError']
+}
+
+export interface TimeoutErrorMetricsEvent {
+  apiId: ApiId
+  labels: Record<string, string>
+  '@type': typeof MetricsEventType['TimeoutError']
+}
+
+export interface InternalErrorMetricsEvent {
```

Comment:
This `InternalErrorMetricsEvent` is deprecated. 
We can exclude it from the event list.

Reply:
thanks!

[`7004211` (#17)](https://github.com/bucketeer-io/javascript-client-sdk/pull/17/commits/700421107c7421f925290adbacac77695f580421)